Object basics: Clearer wording/example with mutations & references (#29846)
TheOdinProject/curriculum: foundations/javascript_basics/object_basics.md

<div class="lesson-note" markdown="1">

> #### Mutating the properties of an object vs. reassigning the object variables

Reviewer: I think the new heading is a little verbose for a heading
```suggestion
#### Mutation vs. reassignment
```

> When you have multiple variables that reference the same object, mutating the properties of that object will affect all the variables that reference it. However, reassigning the object variable to a new object will only change the value of that specific variable, and will not affect the other variables that previously referenced the original object. For example:

Reviewer: Just some small wording tweak given the rest of the review comments.
```suggestion
When multiple variables reference the same object, mutating that object will affect all the variables that reference it. However, reassigning a variable to a new object will only change the value of that specific variable, and will not automatically reassign any other variable that referenced the original object. For example:
```

```javascript
let animal = { species: "dog" };
let dog = animal;

> // Mutating the properties of the object that both 'animal' and 'dog' reference

Reviewer: The object is being mutated, not the property (which is actually being reassigned)
```suggestion
// Mutating the object that both 'animal' and 'dog' reference
```

animal.species = "cat";

console.log(animal); // { species: "cat" }
console.log(dog); // { species: "cat" }

// Reassigning the 'animal' variable to a new object
animal = { species: "bird" };

console.log(animal); // { species: "bird" }
console.log(dog); // { species: "cat" }
```

> In the first part, when we modify the species property of the object that both `animal` and `dog` reference, the change is reflected in both variables. In the second part, when we reassign the `animal` variable to a new object, the `dog` variable still references the original object, while `animal` now points to the new object.

Reviewer: Fixing with the same as the previous review comment, and a small re-wording to accommodate that. Strength added to the `mutate` and `reassign` keywords.
```suggestion
In the first part, there is only one object which is referenced by both `animal` and `dog`. When we **mutate** that object, the change is reflected in both variables. In the second part, when we **reassign** the `animal` variable to a new object; the `dog` variable still references the original object, while `animal` now points to the new object.
```
